front/app/modules/commercial/native_surveys/components/NativeSurveyForm/NativeSurveyForm.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import { JsonForms } from '@jsonforms/react';
+
+// components
+import { Box, Title } from '@citizenlab/cl2-component-library';
+
+// renderers
+import CLCategoryLayout, {
+  clCategoryTester,
+} from 'components/Form/Components/Layouts/CLCategoryLayout';
+import OrderedLayout, {
+  orderedLayoutTester,
+} from 'components/Form/Components/Layouts/OrderedLayout';
+import InputControl, {
+  inputControlTester,
+} from 'components/Form/Components/Controls/InputControl';
+import TextAreaControl, {
+  textAreaControlTester,
+} from 'components/Form/Components/Controls/TextAreaControl';
+import WYSIWYGControl, {
+  WYSIWYGControlTester,
+} from 'components/Form/Components/Controls/WYSIWYGControl';
+import TopicsControl, {
+  topicsControlTester,
+} from 'components/Form/Components/Controls/TopicsControl';
+import ImageControl, {
+  imageControlTester,
+} from 'components/Form/Components/Controls/ImageControl';
+import AttachmentsControl, {
+  attachmentsControlTester,
+} from 'components/Form/Components/Controls/AttachmentsControl';
+import DescriptionControl, {
+  descriptionControlTester,
+} from 'components/Form/Components/Controls/DescriptionControl';
+import TitleControl, {
+  titleControlTester,
+} from 'components/Form/Components/Controls/TitleControl';
+import SingleSelectControl, {
+  singleSelectControlTester,
+} from 'components/Form/Components/Controls/SingleSelectControl';
+import MultiSelectControl, {
+  multiSelectControlTester,
+} from 'components/Form/Components/Controls/MultiSelectControl';
+import UserPickerControl, {
+  userPickerControlTester,
+} from 'components/Form/Components/Controls/UserPickerControl';
+import CheckboxControl, {
+  checkboxControlTester,
+} from 'components/Form/Components/Controls/CheckboxControl';
+import LocationControl, {
+  locationControlTester,
+} from 'components/Form/Components/Controls/LocationControl';
+import DateControl, {
+  dateControlTester,
+} from 'components/Form/Components/Controls/DateControl';
+import MultilocInputLayout, {
+  multilocInputTester,
+} from 'components/Form/Components/Controls/MultilocInputLayout';
+import useLocale from 'hooks/useLocale';
+import { isNilOrError } from 'utils/helperUtils';
+
+const NativeSurveyForm = () => {
+  const locale = useLocale();
+
+  const renderers = [
```

Comment:
Non blocking: We could move these out of the component and keep the component simple to read and follow. In some extreme cases those can even be moved to a different file. That file would only export the renderers. Both of these would be easy to read and follow.